backend/de.metas.business/src/main/java/org/adempiere/acct/api/impl/FactAcctBL.java

```diff
@@ -196,4 +200,30 @@ public void updateFactLineFromDimension(final I_Fact_Acct fa, final AccountDimen
 			fa.setUserElementString7(dim.getUserElementString7());
 		}
 	}
+
+	public void setC_BPartner_ID(@NonNull final I_Fact_Acct factAcct, @Nullable final BPartnerId bpartnerId)
```

Comment:
loud thinking: why not merge `setC_BPartner_ID` and `setC_BPartner_Location_ID` into one method called `setBPartnerAndLocationId(bpartnerLocationId, bpartnerId)` which does something like:

* if bpartnerLocationId is not null
    *  assert bpartnerId is matching if not null
    * set C_BPartner_ID, C_BPartner_Location_ID
* else // if bpartnerLocationId is null
    * if bpartnerId is not null
         *  set C_BPartner_ID, clear C_BPartner_Location_ID
    * else // if bpartner is null
         *  clear C_BPartner_ID, clear C_BPartner_Location_ID


wdyt?

PS: don't forget the `@Overrides`

Reply:
ahh and more, why can't we have this method in FactLine?

Reply:
> loud thinking: why not merge setC_BPartner_ID and setC_BPartner_Location_ID into one method called setBPartnerAndLocationId(bpartnerLocationId, bpartnerId) which does something like:

I like that, will go with it 👍